src/Crawler.php

```diff
@@ -478,9 +478,11 @@ public function startCrawling(UriInterface|string $baseUrl)
 
         $crawlUrl = CrawlUrl::create($this->baseUrl);
 
-        $this->robotsTxt = $this->createRobotsTxt($crawlUrl->url);
+        if ($this->respectRobots) {
+            $this->robotsTxt = $this->createRobotsTxt($crawlUrl->url);
+        }
 
-        if ($this->robotsTxt->allows((string) $crawlUrl->url, $this->getUserAgent()) ||
+        if (null !== $this->robotsTxt && $this->robotsTxt->allows((string) $crawlUrl->url, $this->getUserAgent()) ||
```

Comment:
To me, this condition is unreadable (and not easy to debug). Could you extract it to a function with a good name, and use early returns there to avoid using `&&` and `||`.

Also don't use Yoda style comparisons, but regular ones.

Reply:
Done :+1: